server/sonar-server/src/main/java/org/sonar/server/issue/ws/SearchResponseLoader.java

```diff
@@ -101,13 +101,17 @@ private void loadComments(Collector collector, DbSession dbSession, SearchRespon
       result.setComments(comments);
       for (IssueChangeDto comment : comments) {
         collector.add(USERS, comment.getUserLogin());
-        if (commentService.canEditOrDelete(comment)) {
+        if (canEditOrDelete(comment)) {
           result.addUpdatableComment(comment.getKey());
         }
       }
     }
   }
 
+  private boolean canEditOrDelete(IssueChangeDto dto) {
+    return userSession.isLoggedIn() && userSession.getLogin().equals(dto.getUserLogin());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) NullPointerException might be thrown as 'getLogin' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)